NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -746,23 +793,26 @@ def MuonsFromLQ(T):
 	recomuoninds = []
 	for n in range(T.nMuon):
 		m = TLorentzVector()
-                m.SetPtEtaPhiM(T.Muon_pt[n],T.Muon_eta[n],T.Muon_phi[n],0)
+                m.SetPtEtaPhiM(T.Muon_pt[n],T.Muon_eta[n],T.Muon_phi[n],T.Muon_mass[n])
 		muons.append(m)
-	for n in range(T.nGenPart):
-		pdg = T.GenPart_pdgId[n]
-		if pdg not in [13,-13]:
-			continue
-		motherIndex = T.GenPart_genPartIdxMother[n]
-
-		motherid = 0
-		if motherIndex>-1:
-			motherid = T.GenPart_pdgId[motherIndex]
-			#print 'pdgId:',pdg,'index:',T.GenParticleMotherIndex[n],'mother pdgId:',T.GenParticlePdgId[motherIndex]
-		if motherid not in [42,-42]:
-			continue
-		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],0.0)
-		genmuons.append(m)
+
+	if isData==False:
+
+		for n in range(T.nGenPart):
+			pdg = T.GenPart_pdgId[n]
+			if pdg not in [13,-13]:
+				continue
+			motherIndex = T.GenPart_genPartIdxMother[n]
+
+			motherid = 0
+			if motherIndex>-1:
+				motherid = T.GenPart_pdgId[motherIndex]
+				#print 'pdgId:',pdg,'index:',T.GenParticleMotherIndex[n],'mother pdgId:',T.GenParticlePdgId[motherIndex]
+			if motherid not in [42,-42]:
+				continue
+			m = TLorentzVector()
+			m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],T.GenPart_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.